subsys/bluetooth/mesh/access.c

```diff
@@ -1504,6 +1504,7 @@ static int element_model_recv(struct bt_mesh_msg_ctx *ctx, struct net_buf_simple
 		return ACCESS_STATUS_MESSAGE_NOT_UNDERSTOOD;
 	}
 
+	ctx->rnd_delay = true;
```

Comment:
IMO this should have a Kconfig option. I may want to use delayable messages for some models but not for all. Resetting this value will be annoying.

Reply:
If to use KConfig option, it will be required to add in all commands in all already implemented model servers. I guess this is even more annoying thing for people to revise what already was implemented, debugged and works well. There is no good solution here.

Reply:
> this is even more annoying thing for people to revise what already was implemented, debugged and works well.

This is why having an option is good because they can get the old behavior by setting this option to `n`.

Reply:
> Yes, but this automatically mandates to add this flag initialization in any other models those require delayable message, for instance in all config server commands.

This is not a mandatory requirement. But yes, I though initially exactly about having such option for each model. It may look ugly from the first glance, but it is the same as having timeout value for each model for synchronous call.

Reply:
We could start with single option and then add separate one for each model if we think it is needed.

Reply:
It is not that difficult to add the option.

Reply:
If disable this initialization, it will break delayable message behavior in all existing models. To add this option properly, it requires a lot of changes.

Reply:
> If disable this initialization, it will break delayable message behavior in all existing models.

Can you elaborate? What will break?

Reply:
```
CONFIG_BT_MESH_ACCESS_DELAYABLE_MSG=y
CONFIG_BT_MESH_NEW_REQUESTED_OPTION_TO_INITIALIZE_FLAG_IN_ACCESS=n
```
This configuration combination to not `annoy` hypothetical customers causes the situation when the delayable feature was enabled but actually doesn't work. 
It requires adding this code in any single server command in all servers(SIG models, light models in samples and vendor models). 
```
if (!IS_ENABLED(CONFIG_BT_MESH_NEW_REQUESTED_OPTION_TO_INITIALIZE_FLAG_IN_ACCESS)) {
    ctx->rnd_delay = true;
}
```

Reply:
> This configuration combination to not annoy hypothetical customers causes the situation when the delayable feature was enabled but actually doesn't work.

Not "doesn't work" but not used. You need to manually disable it in your project file. So you know what you are doing.

> It requires adding this code in any single server command in all servers(SIG models, light models in samples and vendor models).

As I said, we could start with `CONFIG_BT_MESH_ACCESS_DELAYABLE_MSG_ENABLED` (like friend, relay, etc. features) which would set `ctx->rnd_delay = true;` in the access layer.